pkg/build/api/types.go

```diff
@@ -201,6 +204,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath
```

Comment:
is it an equivalent of `ContextDir` ? Maybe we should use consistent terminology.